### ISSUE TITLE
Scripts: How to escape paths for command line

### ISSUE THREAD

**Issue by mstv** [gitextensions/gitextensions#11798]
Before the changes were made for `SelectedFiles` to return a blob path rather than full-path I've been making changes so I could concat/replace/etc these variables via "Run on PowerShell" option and one of the problems I ran into is that this kind of escaping did not work.

So I went to see how `ProcessStartInfo` does it and it was different than current method. When I changed current method to what's used in the BCL, then the test-cases for WSL and Windows became identical. Could you look at this commit to see what I mean: https://github.com/mdonatas/gitextensions/commit/768ca88c37ca98ef17e448db6656031eab9718c5

Basically I think we can unify these two methods esp. given that current method did not work when running with PowerShell mode.

_Originally posted by @mdonatas-trafi in https://github.com/gitextensions/gitextensions/issues/11795#issuecomment-2198549641_

**Comment by mstv** [comment]
@mdonatas, please come to an agreement with @SlugFiller. I do not have a strong opinion myself because I hardly encounter files which need escaping.

**Comment by SlugFiller** [comment]
The version I made is based on my experiments on what actually executes, e.g. when passed as parameter to `cmd /c`. If the patch checks out in practice (i.e. Executing with `System.Diagnostics.Process`), then I have no nitpicks about the specifics of the escaped output.
